core/services/workflows/engine.go

```diff
@@ -58,26 +46,32 @@ func (e *Engine) init(ctx context.Context) {
 	retrySec := 5
 	ticker := time.NewTicker(time.Duration(retrySec) * time.Second)
 	defer ticker.Stop()
+
+	trigger := e.workflow.Triggers[0]
```

Comment:
nit: add a "NOTE" comment that we only support the hardcoded workflow with a single trigger